tofhir-engine/src/main/scala/io/tofhir/engine/execution/RunningJobRegistry.scala

```diff
@@ -35,6 +43,8 @@ class RunningJobRegistry(spark: SparkSession) {
    */
   def registerStreamingQuery(execution: FhirMappingJobExecution, mappingUrl: String, streamingQueryFuture: Future[StreamingQuery], blocking: Boolean = false): Future[Unit] = {
     Future {
+      // If there is an error in the streaming query, stop the mapping execution.
+      streamingQueryFuture.recover(_ => stopMappingExecution(execution.job.id, execution.id, mappingUrl))
```

Comment:
Why is this recovered here? Is it because to conveniently call stopMappingExecution function of the registry?

Reply:
Let's assume there is an error such as invalid folder path while running the streaming job which actually cancels the job i.e. the mapping job is not running. However, since this error is unknown to the registry, it assumes that it is still running, therefore we can not rerun it. I made this update to stop mapping job in the registry so that we can rerun it.